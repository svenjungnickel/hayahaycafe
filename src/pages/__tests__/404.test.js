import React from 'react';
import renderer from 'react-test-renderer';
import NotFoundPage from '../404';

jest.mock('../../components/Layout');
jest.mock('../../components/Header');

describe('404', () => {
    it('renders without data throws prop type error', () => {
        const renderNotFoundPage = () => {
            const component = <NotFoundPage />;
            renderer.create(component);
        };

        expect(renderNotFoundPage).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid data throws prop type error', () => {
        const renderNotFoundPage = () => {
            const data = {};

            const component = <NotFoundPage data={data} />;
            renderer.create(component);
        };

        expect(renderNotFoundPage).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const data = {
            headerImage: 'headerImage',
        };

        const component = <NotFoundPage data={data} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
