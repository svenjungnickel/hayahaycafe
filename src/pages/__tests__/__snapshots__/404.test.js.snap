// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 renders correctly 1`] = `
<main>
  <header />
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h1>
            404: not found
          </h1>
          <p>
            You just hit a route that doesn't exist.
          </p>
        </div>
      </div>
    </div>
  </section>
</main>
`;
